tvm/src/pass/stream_inference.cc

```diff
@@ -120,11 +131,25 @@ class NewChannelGathers final : public IRMutator {
         CHECK(new_var.defined());
         target_load_expr = e;
         target_load_access_indices.push_back(op->index);
+
+        if (inside_select_node) {
+            HCL_DEBUG_LEVEL(2) << "[debug] Hmm.. Found load inside the select expr...";
+            insert_conditional_load = true;
+        }
         return Load::make(op->type, new_var, 0, op->predicate);
     }
     return IRMutator::Mutate_(op, e);
   }
 
+  // Handle the load inside Select case 
+  Expr Mutate_(const Select* op, const Expr& e) {
+    inside_select_node = true;
```

Comment:
I think only "load" in two branches needs to be done the if-else conversion, otherwise ("load" in condition) may not be converted.

Reply:
Moreover, the original ternary instructions also need to be eliminated.

Reply:
I do not quite get what you mean. Here I am just adding a conditional read on the the FIFO consumer side:
```c++
B_temp = select(i < 10, B_pipe, 0)
C[i] = select((i< 10), B_temp, some expressions)
```
And all the select nodes will be realized as if-then-else statements in the code generation. So we do not do any actual if-then-else conversion here.

Reply:
@chhzh123 Fixed now. We will generate a merged IfTheElse statement in this case.